modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb

```diff
@@ -109,21 +110,40 @@ def individual_page1_signatures(power_of_attorney, first_name, last_name)
         ]
       end
 
-      def individual_page2_signatures(power_of_attorney, first_name, last_name)
+      def individual_page2_signatures(power_of_attorney, rep_first_name, rep_last_name)
+        first_name, last_name = veteran_or_claimant_signature(power_of_attorney)
         [
           {
-            'signature' => "#{power_of_attorney.auth_headers['va_eauth_firstName']} " \
-                           "#{power_of_attorney.auth_headers['va_eauth_lastName']} - signed via api.va.gov",
+            'signature' => "#{first_name} " \
+                           "#{last_name} - signed via api.va.gov",
             'x' => 35,
             'y' => 306
           },
           {
-            'signature' => "#{first_name} #{last_name} - signed via api.va.gov",
+            'signature' => "#{rep_first_name} #{rep_last_name} - signed via api.va.gov",
             'x' => 35,
             'y' => 200
           }
         ]
       end
+
+      def veteran_or_claimant_signature(power_of_attorney)
+        claimant_icn = power_of_attorney.form_data['claimant']['claimantId']
```

Comment:
I know the ticket says `claimantId` is optional. We should double check how to handle requests that have claimant attributes but no `claimantId` - are we comfortable e-signing these as a Veteran or should the requests get rejected?

Reply:
Logic added to raise 422 if claimant attributes present but no claimantId.